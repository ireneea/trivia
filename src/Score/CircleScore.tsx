import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { colors, fontSize } from "../styles/globals";

type Props = {
  value: number;
  radius: number;
  valueSize?: number;
  labelSize?: number;
  label: string;
  borderColor?: string;
  backgroundColor?: string;
  textColor?: string;
  testID?: string;
};

const CircleScore: React.FC<Props> = (props) => {
  const {
    value,
    label,
    radius,
    valueSize = 20,
    labelSize = 20,
    borderColor = colors.background,
    backgroundColor = colors.primary,
    textColor = colors.background,
    testID,
  } = props;
  return (
    <View style={{ ...styles.container, width: radius * 2, height: radius * 2, backgroundColor, borderColor }}>
      <Text style={{ fontSize: valueSize, color: textColor, fontWeight: "500" }} testID={testID}>
        {value}
      </Text>
      <Text style={{ ...styles.label, fontSize: labelSize, color: textColor }}>{label}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: 1000,
    borderWidth: 5,
    borderStyle: "solid",
    justifyContent: "center",
    alignItems: "center",
  },
  label: {
    textTransform: "uppercase",
  },
});

export default CircleScore;
