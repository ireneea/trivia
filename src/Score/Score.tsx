import React from "react";
import { View, Text, TouchableWithoutFeedback } from "react-native";
import { globalStyles } from "../styles/globals";

type Props = {
  navigation: any;
};

const Score: React.FC<Props> = (props) => {
  const navigateTo = (page: string) => {
    const { navigation } = props;
    navigation.navigate(page);
  };

  return (
    <View testID="scoreScreen" style={globalStyles.centredContainer}>
      <TouchableWithoutFeedback
        testID="newGameBtn"
        onPress={() => navigateTo("Game")}
      >
        <Text>New Game</Text>
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback
        testID="homeBtn"
        onPress={() => navigateTo("Start")}
      >
        <Text>Home</Text>
      </TouchableWithoutFeedback>
    </View>
  );
};

export default Score;
