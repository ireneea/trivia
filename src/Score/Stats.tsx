import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { colors, fontSize } from "../styles/globals";

type StatElementProps = {
  value: number;
  name: string;
};

const StatElement: React.FC<StatElementProps> = (props) => {
  const { value, name } = props;
  return (
    <View style={statElementStyles.container}>
      <Text testID={name} style={statElementStyles.valueText}>
        {value}
      </Text>
      <Text style={statElementStyles.nameText}>{name}</Text>
    </View>
  );
};

const statElementStyles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
  },
  valueText: {
    color: colors.normal,
    ...fontSize.xl,
  },
  nameText: {
    color: colors.secondary,
    textTransform: "uppercase",
    marginTop: 5,
    ...fontSize.s,
  },
});

type Props = {
  correct: number;
  incorrect: number;
  unanswered: number;
};

const Stats: React.FC<Props> = (props) => {
  const { correct, incorrect, unanswered } = props;

  return (
    <View style={statsStyles.statsContainer}>
      <StatElement value={correct} name="correct" />
      <StatElement value={incorrect} name="incorrect" />
      <StatElement value={unanswered} name="unanswered" />
    </View>
  );
};

const statsStyles = StyleSheet.create({
  statsContainer: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between",
    width: "100%",
    marginTop: 20,
  },
});

export default Stats;
