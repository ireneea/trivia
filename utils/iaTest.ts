import { ReactTestInstance } from "react-test-renderer";

export function findByTestID(
  root: ReactTestInstance,
  testID: string
): ReactTestInstance {
  return root.findByProps({ testID });
}

export function mockNavigationProps(props = {}) {
  return {
    navigation: {
      // when use in a unit test jest will be defined
      navigate: jest.fn(),
    },
    ...props,
  };
}
