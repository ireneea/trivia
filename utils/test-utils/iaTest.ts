export function mockNavigationProps(props: Object = {}): Object {
  return {
    navigation: {
      // when use in a unit test jest will be defined
      navigate: jest.fn(),
    },
    ...props,
  };
}

export function mockRouteParamsProps(params: Object, props: Object = {}): Object {
  return {
    route: {
      params,
    },
    ...props,
  };
}
