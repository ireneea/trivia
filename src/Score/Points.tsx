import React from "react";
import { View, Text, StyleSheet, Dimensions } from "react-native";

import { globalStyles, colors, fontSize } from "../styles/globals";

import PercentageCircle from "./PercentageCircle";
import CircleScore from "./CircleScore";

type Props = {
  points: number;
  bonus: number;
  streak: number;
};

const Points: React.FC<Props> = (props) => {
  const { points, bonus, streak } = props;

  const xCentre = Math.round(Dimensions.get("window").width) / 2;

  return (
    <View style={styles.container}>
      <View style={{ position: "absolute", left: xCentre - 30, top: 310 }}>
        <CircleScore
          value={streak}
          label="Highest streak"
          radius={80}
          valueSize={49}
          labelSize={13}
          borderColor={colors.primary}
          backgroundColor={colors.background}
          textColor={colors.primary}
          testID="streak"
        />
      </View>
      <View style={{ position: "absolute", left: xCentre - 80, top: 100 }}>
        <CircleScore value={points} label="Total Score" radius={120} valueSize={76} labelSize={20} testID="points" />
      </View>
      <View style={{ position: "absolute", left: xCentre - 150, top: 260 }}>
        <CircleScore value={bonus} label="Bonus Points" radius={75} valueSize={49} labelSize={13} testID="bonus" />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    ...globalStyles.centredContainer,
    // flex: 1,
  },
  pointsText: {
    marginTop: 20,
    color: colors.primary,
    ...fontSize.xxxxl,
  },
  accuracyContainer: {
    marginTop: 20,
  },
  accuracyText: {
    color: colors.normal,
    ...fontSize.xl,
  },
});

export default Points;
