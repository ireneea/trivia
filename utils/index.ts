import * as test from "./iaTest";

export default {
  test,
};
