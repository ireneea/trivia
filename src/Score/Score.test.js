import React from "react";
import { render, fireEvent } from "@testing-library/react-native";

import Score from "./Score";
import { mockNavigationProps } from "../../utils/iaTest";

describe("Score", () => {
  it("render Score correctly", () => {
    const { baseElement } = render(<Score />);
    expect(baseElement).toMatchSnapshot();
  });

  it("navigate to Game", async () => {
    const props = mockNavigationProps();

    const { getByText } = render(<Score {...props} />);
    fireEvent.press(getByText("New Game"));

    expect(props.navigation.navigate).toHaveBeenCalledWith("Game");
  });

  it("navigate to Start", () => {
    const props = mockNavigationProps();

    const { getByText } = render(<Score {...props} />);
    fireEvent.press(getByText("Home"));

    expect(props.navigation.navigate).toHaveBeenCalledWith("Start");
  });
});
