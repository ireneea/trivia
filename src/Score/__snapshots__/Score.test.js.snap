// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score render Score correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID="scoreScreen"
    >
      <TouchableWithoutFeedback
        testID="newGameBtn"
      >
        <Text>
          New Game
        </Text>
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback
        testID="homeBtn"
      >
        <Text>
          Home
        </Text>
      </TouchableWithoutFeedback>
    </View>
  </View>
</View>
`;
